app/javascript/gobierto_visualizations/webapp/visualizations/treeMapNested.vue

```diff
@@ -377,39 +383,55 @@ export default {
           })
 
         g.selectAll(".foreignobj")
-        .on('mousemove', function(d) {
+        .on('mousemove', function(d, i) {
           const { depth } = d
           if (depth === 1) return;
           let title = d.data.name === undefined ? d.data.title : d.data.name;
-          const coordinates = d3.mouse(this);
-          const x = coordinates[0];
-          const y = coordinates[1];
+          const [x, y] = d3.mouse(this);
 
           //Elements to determinate the position of tooltip
           const container = document.getElementsByClassName('container-tree-map-nested')[0];
           const containerWidth = container.offsetWidth
-          const tooltipWidth = tooltip.node().offsetWidth
+          const tooltipWidth = depth === 2 ? tooltipAssignee.node().offsetWidth : tooltip.node().offsetWidth
+          const tooltipHeight = depth === 2 ? tooltipAssignee.node().offsetHeight : tooltip.node().offsetHeight
           const positionWidthTooltip = x + tooltipWidth
-          const positionTop = `${y - 20}px`
-          const positionLeft = `${x + 10}px`
-          const positionRight = `${x - tooltipWidth - 10}px`
+          const positionRight = `${x - tooltipWidth - 20}px`
+          const windowHeight = window.innerHeight
+          const positionTop = tooltipHeight + y > windowHeight ? `${y - (tooltipHeight / 2)}px` : `${y}px`
+          const positionLeft = `${x + 20}px`
 
           if (depth === 2) {
-            let labelTotalContracts = `${I18n.t('gobierto_visualizations.visualizations.contracts.contracts')}`
             let totalContracts = d.children === undefined ? '' : d.children
+            let totalContractsLength
+            let contractsString = ''
             if (totalContracts) {
-              totalContracts = totalContracts.filter(contract => typeof contract.data !== "function").length
-              labelTotalContracts = totalContracts > 1 ? labelTotalContracts : `${I18n.t('gobierto_visualizations.visualizations.contracts.contract')}`
+              totalContracts = totalContracts.filter(contract => typeof contract.data !== "function")
+              totalContractsLength = totalContracts.length
```

Comment:
Do you actually need to declare this?

Reply:
Yes

Reply:
Sure? The only use of `totalContractsLength` is in the `while`
So: `while (i < totalContracts.length) {`